openmc/mgxs/mgxs.py

```diff
@@ -4809,64 +4849,953 @@ class Chi(MGXS):
+                if self.domain_type == 'mesh':
+                    domain_filter = filter_type(self.domain)
+                else:
+                    domain_filter = filter_type(self.domain.id)
+
+                # Create each Tally needed to compute the multi group cross section
+                self._tallies[tally_key] = openmc.Tally(name=self.name)
+                self._tallies[tally_key].estimator = 'analog'
+                self._tallies[tally_key].scores = [self.scores[-1]]
+                self._tallies[tally_key].filters = [domain_filter]
+                self._tallies[tally_key].filters.extend(self.filters[-1])
+
+                # If a tally trigger was specified, add it to each tally
+                if self.tally_trigger:
+                    trigger_clone = copy.deepcopy(self.tally_trigger)
+                    trigger_clone.scores = [self.scores[-1]]
+                    self._tallies[tally_key].triggers.append(trigger_clone)
+
+                # If this is a by-nuclide cross-section, add nuclides to Tally
+                if self.by_nuclide:
+                    self._tallies[tally_key].nuclides += self.get_nuclides()
+                else:
+                    self._tallies[tally_key].nuclides.append('total')
+
+        return self._tallies
+
+    @property
+    def scatter_xs(self):
+        return self.mgxs[0]
+
+    @property
+    def probability_matrix(self):
+        return self.mgxs[1]
+
+    @property
+    def multiplicity_matrix(self):
+        return self.mgxs[2]
+
+    @property
+    def rxn_rate_tally(self):
+        raise NotImplementedError('The reaction rate tally is not well defined ')
+
+    @property
+    def xs_tally(self):
+        if self._xs_tally is None:
+            if self.tallies is None:
+                msg = 'Unable to get xs_tally since tallies have ' \
+                      'not been loaded from a statepoint'
+                raise ValueError(msg)
+
+            # If using P0 correction subtract scatter-1 from the diagonal
+            if self.correction == 'P0' and self.legendre_order == 0:
+                flux = self.tallies['{} : flux'.format(self.rxn_type)]
+                scatter_p0 = self.tallies['{0} : {0}'.format(self.rxn_type)]
+                scatter_p1 = self.tallies['{}-1'.format(self.rxn_type)]
+
+                energy_filter = scatter_p0.find_filter(openmc.EnergyFilter)
+                energy_filter = copy.deepcopy(energy_filter)
+                scatter_p1 = scatter_p1.diagonalize_filter(energy_filter)
+                correction = scatter_p1 / flux
+            else:
+                correction = 0.
+
+            self._xs_tally = \
+                self.scatter_xs.xs_tally * self.probability_matrix.xs_tally
+
+            if self.nu:
+                self._xs_tally *= self.multiplicity_matrix.xs_tally
+
+            self._xs_tally -= correction
+            self._compute_xs()
+
+        return self._xs_tally
+
+    @ScatterMatrixXS.correction.setter
```

Comment:
That's right. I actually realized that these inherited properties weren't implemented correctly, and will be pushing the fix in a moment.